src/prism.c

```diff
@@ -11176,8 +11176,14 @@ parse_arguments(pm_parser_t *parser, pm_arguments_t *arguments, bool accepts_for
 
                 if (token_begins_expression_p(parser->current.type)) {
                     expression = parse_value_expression(parser, PM_BINDING_POWER_DEFINED, PM_ERR_EXPECT_ARGUMENT);
-                } else if (pm_parser_local_depth(parser, &operator) == -1) {
-                    pm_parser_err_token(parser, &operator, PM_ERR_ARGUMENT_NO_FORWARDING_AMP);
+                } else {
+                    if (pm_parser_local_depth(parser, &operator) == -1) {
```

Comment:
What do you think about adding `&` to the local table instead of checking it here? That would make it so that this first check would work.

Reply:
Please see https://github.com/ruby/prism/issues/1839#issuecomment-1822133511.